### ISSUE TITLE
Adding non primitive  loss functions

### ISSUE THREAD

**Issue by MaharshiYeluri01** [PyTorchLightning/pytorch-lightning#1914]
It would be great if there are some readily available non primitive loss functions like
AngularLoss 
ArcFaceLoss 
CircleLoss 
TripletLoss 
LargeMarginSoftmaxLoss
MarginLoss ......
......

**Comment by MaharshiYeluri01** [comment]
https://pypi.org/project/pytorch-metric-learning/
